exporters/metrics/src/main/java/com.google.cloud.opentelemetry.metric/MetricExporter.java

```diff
@@ -0,0 +1,205 @@
+    MetricServiceSettings.Builder builder =
+        MetricServiceSettings.newBuilder()
+            .setCredentialsProvider(
+                FixedCredentialsProvider.create(checkNotNull(credentials, "Credentials not provided.")));
+    builder.createMetricDescriptorSettings()
+        .setSimpleTimeoutNoRetries(org.threeten.bp.Duration.ofMillis(deadline.toMillis()));
+    return new MetricExporter(projectId, MetricServiceClient.create(builder.build()), false);
+  }
+
+  @Override
+  public ResultCode export(Collection<MetricData> metrics) {
+    List<TimeSeries> allTimesSeries = new ArrayList<>();
+
+    for (MetricData metricData : metrics) {
+      MetricDescriptor descriptor = mapMetricDescriptor(metricData, uniqueIdentifier);
+      if (descriptor == null) {
+        continue;
+      }
+      metricServiceClient.createMetricDescriptor(
+          CreateMetricDescriptorRequest.newBuilder().setName(PROJECT_NAME_PREFIX + projectId)
+              .setMetricDescriptor(descriptor)
+              .build());
+
+      MetricWithLabels updateKey = new MetricWithLabels(descriptor.getType(),
+          metricData.getDescriptor().getConstantLabels());
+
+      // We are expecting one point per MetricData
+      Optional<MetricData.Point> metricPoint = metricData.getPoints().stream()
+          .reduce((first, second) -> second);
+      if (!metricPoint.isPresent()) {
+        logger.log(WARNING, "No point found in metric {0}", metricData);
+        continue;
+      }
+
+      // Cloud Monitoring API allows, for any combination of labels and
+      // metric name, one update per WRITE_INTERVAL seconds
+      long pointCollectionTime = metricPoint.get().getEpochNanos();
+      if (lastUpdatedTime.containsKey(updateKey)
+          && pointCollectionTime <= lastUpdatedTime.get(updateKey) / NANO_PER_SECOND + WRITE_INTERVAL_SECOND) {
+        continue;
+      }
+
+      Metric metric = mapMetric(metricData, descriptor.getType(), uniqueIdentifier);
+      Point point = mapPoint(lastUpdatedTime, metricData, updateKey, exporterStartTime, pointCollectionTime);
+      if (point == null) {
+        continue;
+      }
+
+      allTimesSeries.add(TimeSeries.newBuilder()
+          .setMetric(metric)
+          .addPoints(point)
+          .setResource(mapGcpMonitoredResource(metricData.getResource()))
+          .build());
+    }
+    createTimeSeriesBatch(metricServiceClient, ProjectName.of(projectId), allTimesSeries);
+    return ResultCode.SUCCESS;
+  }
+
+  private static void createTimeSeriesBatch(MetricServiceClient metricServiceClient, ProjectName projectName,
+      List<TimeSeries> allTimesSeries) {
+    List<List<TimeSeries>> batches = Lists.partition(allTimesSeries, MAX_BATCH_SIZE);
+    for (List<TimeSeries> timeSeries : batches) {
+      metricServiceClient.createTimeSeries(projectName, new ArrayList<>(timeSeries));
+    }
+  }
+
+  @Override
+  public ResultCode flush() {
+    // TODO (zoe): add support for flush
```

Comment:
Could you change the TODO to refer to your GitHub handle please?
Also, isn't this a blocker for using the exporter?

Reply:
I've changed this to return `SUCCESS` always based on https://github.com/open-telemetry/opentelemetry-java/blob/master/exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcMetricExporter.java#L136